lib/logger/logger.go

```diff
@@ -130,16 +129,60 @@ func logLevelSkipframes(skipframes int, level, format string, args ...interface{
 func errorsAndWarnsLoggedCleaner() {
 	for {
 		time.Sleep(time.Second)
-		atomic.StoreUint64(&errorsLogged, 0)
-		atomic.StoreUint64(&warnsLogged, 0)
+		logLimiter.reset()
 	}
 }
 
 var (
-	errorsLogged uint64
-	warnsLogged  uint64
+	logLimiter = newLogLimit()
 )
 
+func newLogLimit() logLimit {
```

Comment:
I guess, that go vet reports error, if we pass something with mutex by its value with mutex copy. In this case, its safe to return struct instead of pointer, but i agree, better to change it to pointer. Fixed.